### ISSUE TITLE
[Bug] Fluid Interaction generates block at a wrong pos.

### ISSUE THREAD

**Issue by Yesterday17** [DaedalusGame/Alfinivia#2]
When it comes to a user defined interaction, it acts like this:
![2018-05-25_15 51 04](https://user-images.githubusercontent.com/8667822/40553861-8bb43aba-6076-11e8-9fe0-f8273dd134aa.png)

But the correct result should be like this:
![2018-05-25_15 52 05](https://user-images.githubusercontent.com/8667822/40553921-bf4f70ba-6076-11e8-9d6f-6ea52e1ab3e8.png)

(ignore the strange blocks nearby plz (xd

**Comment by DaedalusGame** [comment]
right, it's always been sort of wierd how it behaved. I will have another look at it. maybe i can make it work a little better

**Comment by DaedalusGame** [comment]
Out of curiosity, does it work properly if you flip the fluids in your script?

**Comment by Yesterday17** [comment]
@DaedalusGame 
Seems working well...
![2018-05-26_01 48 56](https://user-images.githubusercontent.com/8667822/40571380-38f5e482-60ca-11e8-9add-3f1914281219.png)

But if you exchange them, it works like this:
![2018-05-26_01 49 36](https://user-images.githubusercontent.com/8667822/40571386-563c4270-60ca-11e8-9c21-a10a8ac5cff0.png)

This effect also happens when you place the two liquids like this:
Steps:
1. Place quicksand(the liquid defined below).
2. Place water.

Result:
![2018-05-26_01 58 14](https://user-images.githubusercontent.com/8667822/40571436-57d06a5c-60cb-11e8-9eef-2249a1f31496.png)
I guess it's related to the material of one liquid, but it seems not true.

One of the liquid is `<liquid:water>`, and [the other one](https://github.com/Yesterday17/FluidVoid/blob/master/scripts/fluid/quicksand.zs) is defined by ContentTweaker.
```
#priority 500
#loader contenttweaker


import scripts.utils.fluid.f_default;
import scripts.utils.fluid.f_viscosity;
import scripts.utils.fluid.f_density;


var ct_quicksand = f_default("ct_quicksand", "CC9933");


# It's almost impossible to walk in quicksand.
f_density(ct_quicksand, 1);


# The liquid spreads slower than water.
f_viscosity(ct_quicksand, 20);


ct_quicksand.register();
```

**Comment by DaedalusGame** [comment]
right. so i can't blame this on a special secret system. it's just funky in general. i will take a look soon
